src/tests/dictionaryConcepts/container/SelectAnswers.test.js

```diff
@@ -13,22 +17,58 @@ describe('<SelectAnswers />', () => {
       source: 'test source',
       frontEndUniqueKey: 'unique',
     };
-    wrapper = shallow(<SelectAnswers {...props} />);
+    wrapper = mount(<SelectAnswers {...props} />);
   });
 
-  it('should handle change in AsncSelect', () => {
-    const instance = wrapper.instance();
-    const spy = jest.spyOn(instance, 'handleChange');
-    instance.handleChange(['value']);
+  it('should handle key down event when a user presses enter button to search for concept', async () => {
+    const spy = jest.spyOn(wrapper.instance(), 'handleKeyPress');
+    const event = { keyCode: KEY_CODE_FOR_ENTER };
+    await wrapper.instance().handleKeyPress(event, 'malaria');
     expect(spy).toHaveBeenCalled();
-    instance.handleChange(null);
+  });
+
+  it('should handle key down event when a user presses enter button to search for concept with input less that 3', async () => {
+    const event = { keyCode: KEY_CODE_FOR_ENTER };
+    await wrapper.instance().handleKeyPress(event, 'ma');
+    expect(wrapper.state().isVisible).toBeFalsy();
+  });
+
+  it('should handle key down event when a user presses button thats not enter to search for concept', () => {
+    const spy = jest.spyOn(wrapper.instance(), 'handleKeyPress');
+    const event = { keyCode: KEY_CODE_FOR_SPACE };
+    wrapper.find('#searchInputCiel').simulate('keyDown', event);
     expect(spy).toHaveBeenCalled();
   });
 
-  it('should update the state with current input value', () => {
-    wrapper = mount(<SelectAnswers {...props} />);
-    const instance = wrapper.instance();
-    instance.handleInputChange('searchTerm');
-    expect(wrapper.state().inputValue).toEqual('searchTerm');
+  it('should handle key down event when a user presses escape button to remove the dropdown', () => {
+    const spy = jest.spyOn(wrapper.instance(), 'handleKeyPress');
+    const event = { keyCode: KEY_CODE_FOR_ESCAPE };
+    wrapper.find('#searchInputCiel').simulate('keyDown', event);
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('should handle key down event when a user presses button thats not enter to search for concept', () => {
```

Comment:
Which user interface button are we pressing here? Why should a button press result into a key down event?

Reply:
I have updated the test description to `should handle click when a user clicks to select a prefered concept` since the test is meant to handle the interface where a user is clicking to select a given concept.